blockchain/blockdao.go

```diff
@@ -521,6 +523,11 @@ func (dao *blockDAO) putBlock(blk *Block) error {
 		batch.Put(blockNS, topHeightKey, height, "failed to put top height")
 	}
 
+	// only log this info if you enable explorer
+	if !dao.config.Explorer.Enabled {
+		return batch.Commit()
+	}
+
 	value, err = dao.kvstore.Get(blockNS, totalTransfersKey)
```

Comment:
also i think we should plan to create a separate db file for these index?
so the 'blockchain' db file is identical for all nodes, and node with explorer enabled has an extra index db

Reply:
@dustinxie that's my second step. In first step, I just separate the code.